saleor/core/auth_backend.py

```diff
@@ -55,7 +57,8 @@ class PluginBackend(JSONWebTokenBackend):
     def authenticate(self, request=None, **kwargs):
         if request is None:
             return None
-        return request.plugins.authenticate_user(request)
+        manager = PluginsManager(settings.PLUGINS)
```

Comment:
I am not conviced about this. The PluginBackend is called after we tried to fetch user by `JSONWebTokenBackend`, if it can't find a userl, then `PluginBackend` will be called. It means that for each non-authorized request we will fetch all manager's details. It means -/+ 7 db queries for each request, and we don't use dataloader here, so it means that call dataloader for plugin manager will give us 7 additionally DB queries.

Reply:
How exactly does a dataloader work in a request life cycle? If authentication is done once per request, that means so many times would this particular PluginsManager be called here. Please note that particular PluginManager does not use requestor because it needs to provide authentication if JWT bails out. But that makes it unusable in other cases where we need to have a PluginManager that uses requestor. So there's no use for dataloader in this particular place.

Reply:
I made mistake in DB query calculations. It is not 7 but 3. 
With current implementation each anonymous request or with external log-in will add 3 DB queries. It is not a big number from perspective of single request, but if we will take into account  hundreds request per minute, the number is big.
The ideal approach would be change the behaviour of manager to assign requestor in different way than it is currently implemented or something that would allow us to not initialize Manager twice (as it is done now). 
I think that we should cover here at least anonymous request. Instead of initializing PluginsManager manualy, we can call `manager = AnonymousPluginManagerLoader(request).load("Anonymous").get()`
```
class PluginBackend(JSONWebTokenBackend):
    def authenticate(self, request=None, **kwargs):
        if request is None:
            return None
        
        manager = AnonymousPluginManagerLoader(request).load("Anonymous").get()
        return manager.authenticate_user(request)
```
(Anonymous request will use the same dataloader as is used here)
IMO External user could be solved in separate PR (or maybe you have an idea, how to cover this part also?)

Reply:
Yeah, I came up with the same anonymous loader :wink: And after this PR I started working on requestor dataloader that should address what you just mentioned :smile: